test/jdk/java/net/httpclient/ProxyServer.java

```diff
@@ -656,10 +647,11 @@ public static void main(String[] args) throws Exception {
         int port = Integer.parseInt(args[0]);
         boolean debug = args.length > 1 && args[1].equals("-debug");
         System.out.println("Debugging : " + debug);
-        ProxyServer ps = new ProxyServer(port, debug);
-        System.out.println("Proxy server listening on port " + ps.getPort());
-        while (true) {
-            Thread.sleep(5000);
+        try (ProxyServer ps = new ProxyServer(port, debug)) {
```

Comment:
If we want users to properly wrap `ProxyServer` in try-with-resources, we should be such an example too.